ariadne/src/Ariadne/Config/Cardano.hs

```diff
@@ -1,29 +1,333 @@
+    , Cardano.npBehaviorConfig = npBehaviorConfig
+    }
+
+data NetworkConfigOpts = NetworkConfigOpts
+    { ncoTopology :: !(Presence (Topology KademliaParams))
+    , ncoPort  :: !Word16
+    } deriving Show
+
+instance Eq NetworkConfigOpts where
+    nco == nco' = ncoPort nco ==  ncoPort nco'
+
+toCardanoNetworkConfigOpts :: NetworkConfigOpts -> Cardano.NetworkConfigOpts
+toCardanoNetworkConfigOpts NetworkConfigOpts{..} = Cardano.NetworkConfigOpts
+    { Cardano.ncoTopology = ncoTopology ^? _File
+    , Cardano.ncoKademlia = Nothing
+    , Cardano.ncoSelf = Nothing
+    , Cardano.ncoPort = ncoPort
+    , Cardano.ncoPolicies = Nothing
+    , Cardano.ncoBindAddress = Nothing
+    , Cardano.ncoExternalAddress = Nothing
+    }
+
+data ConfigurationOptions = ConfigurationOptions
+    { cfo :: !(Presence Configuration) }
+    deriving (Eq, Show)
+
+instance Default ConfigurationOptions where
+    def = ConfigurationOptions $ There def
+
+toCardanoConfigurationOptions :: ConfigurationOptions -> Cardano.ConfigurationOptions
+toCardanoConfigurationOptions ConfigurationOptions{..} = Cardano.ConfigurationOptions
+    { Cardano.cfoFilePath = maybe "" id $ cfo ^? _File
+    , Cardano.cfoKey = "mainnet_full"
+    , Cardano.cfoSystemStart = Nothing
+    , Cardano.cfoSeed = Nothing
+    }
+
+data CommonArgs = CommonArgs
+    { logConfig            :: !(Presence LoggerConfig)
+    , logPrefix            :: !(Maybe FilePath)
+    , configurationOptions :: !ConfigurationOptions
+    }
+
+-- Checked all fields except logConfig
+instance Eq CommonArgs where
+    ca == ca' = logPrefix ca == logPrefix ca'
+        && configurationOptions ca == configurationOptions ca'
+
+instance Show CommonArgs where
+    show CommonArgs{..} = "CommonArgs { " <> show logPrefix <>
+        ", " <> show configurationOptions <> " }"
+
+toCardanoCommonArgs :: CommonArgs -> Cardano.CommonArgs
+toCardanoCommonArgs CommonArgs{..} = Cardano.CommonArgs
+    { Cardano.logConfig = logConfig ^? _File
+    , Cardano.logPrefix = Nothing
+    , Cardano.reportServers = []
+    , Cardano.updateServers = []
+    , Cardano.configurationOptions = toCardanoConfigurationOptions configurationOptions
+    }
+
+newtype CardanoConfig = CardanoConfig
+    { getCardanoConfig :: CommonNodeArgs }
+    deriving (Eq, Show)
+
+instance D.Interpret CardanoConfig where
+    autoWith _ = CardanoConfig <$> interpretCommonNodeArgs
+
+instance D.Inject CardanoConfig where
+    injectWith _ = contramap getCardanoConfig injectCommonNodeArgs
+
+instance Default RotationParameters where
+    def = RotationParameters
+        { rpLogLimit = 104857600
+        , rpKeepFiles = 20
+        }
+
+instance Default LoggerTree where
+    def = LoggerTree
+        { _ltSubloggers = mempty
+        , _ltFiles =
+            [ HandlerWrap "logs/node.pub" Nothing
+            , HandlerWrap "logs/node" Nothing
+            ]
+        , _ltSeverity = Just debugPlus
+        }
+
+instance Default LoggerConfig where
```

Comment:
Since you need to embed `Configuration` anyway, I'd do the same for `LoggerConfig` (embed `log-config.yaml` and parse it), otherwise it's duplicated.